File: src/server/gkill/api/req_res/update_kc_request.go
package req_res

import "github.com/mt3hr/gkill/src/app/gkill/dao/reps"

type UpdateKCRequest struct {
	SessionID string `json:"session_id"`

	KC reps.KC `json:"kc"`

	TXID *string `json:"tx_id"`

	LocaleName string `json:"locale_name"`

	UpdatedKyou *reps.Kyou `json:"updated_kyou"`

	WantResponseKyou bool `json:"want_response_kyou"`
}

File: src/client/classes/api/req_res/update-kc-request.ts
'use strict'

import { KC } from '@/classes/datas/kc'
import { GkillAPIRequest } from '../gkill-api-request'
import { Kyou } from '@/classes/datas/kyou'

export class UpdateKCRequest extends GkillAPIRequest {

    kc: KC

    tx_id: string | null = null

    want_response_kyou: boolean

    updated_kyou: Kyou | null | null = null

    constructor() {
        super()
        this.kc = new KC()
        this.want_response_kyou = false
    }

}


